components/dashboard/DashboardPagination.vue

```diff
@@ -0,0 +1,32 @@
+<script setup lang="ts">
+import { defineProps } from 'vue';
+
+const props = defineProps<{
+    currentPage: number;
+    totalPages: number;
+    prevPage: () => void;
```

Comment:
This is like react. use emit. when changing a card, do not pass function

Reply:
`<script setup lang="ts">
import { defineProps, defineEmits } from 'vue';

const props = defineProps<{
    currentPage: number;
    totalPages: number;
}>();

const _emit = defineEmits(['prevPage', 'nextPage']);
</script>

<template>
    <div class="flex justify-between items-center mt-4">
        <button
            :disabled="props.currentPage === 1"
            class="px-4 py-2 bg-gray-300 text-gray-700 rounded hover:bg-gray-400 transition-colors duration-300 disabled:opacity-50 disabled:cursor-not-allowed flex items-center"
            @click="_emit('prevPage')">
            Previous
        </button>
        <span class="text-gray-700 font-semibold">Page {{ props.currentPage }} of {{ props.totalPages }}</span>
        <button
            :disabled="props.currentPage === props.totalPages"
            class="px-4 py-2 bg-gray-300 text-gray-700 rounded hover:bg-gray-400 transition-colors duration-300 disabled:opacity-50 disabled:cursor-not-allowed flex items-center"
            @click="_emit('nextPage')">
            Next
        </button>
    </div>
</template>

<style scoped>
@tailwind utilities;
</style>`



<PaginationControls
          :current-page="currentPage"
          :total-pages="totalPages"
          @prev-page="prevPage"
          @next-page="nextPage"
        />

Reply:
![image](https://github.com/user-attachments/assets/42689763-e97f-4b78-a85c-c57de4b67deb)
![image](https://github.com/user-attachments/assets/dc88af2c-e987-410c-a6f6-6d48a4517136)

already resolve the issue here